oauth2_http/java/com/google/auth/oauth2/GoogleCredentials.java

```diff
@@ -185,14 +188,55 @@ public static GoogleCredentials fromStream(
         fileType)) {
       return ExternalAccountAuthorizedUserCredentials.fromJson(fileContents, transportFactory);
     }
-    if ("impersonated_service_account".equals(fileType)) {
+    if (ImpersonatedCredentials.IMPERSONATED_CREDENTIALS_FILE_TYPE.equals(fileType)) {
       return ImpersonatedCredentials.fromJson(fileContents, transportFactory);
     }
     throw new IOException(
         String.format(
             "Error reading credentials from stream, 'type' value '%s' not recognized."
-                + " Expecting '%s' or '%s'.",
-            fileType, USER_FILE_TYPE, SERVICE_ACCOUNT_FILE_TYPE));
+                + " Valid values are '%s', '%s', '%s', '%s', '%s', 's'.",
+            fileType,
+            USER_FILE_TYPE,
+            SERVICE_ACCOUNT_FILE_TYPE,
+            GDCH_SERVICE_ACCOUNT_FILE_TYPE,
+            ExternalAccountCredentials.EXTERNAL_ACCOUNT_FILE_TYPE,
+            ExternalAccountAuthorizedUserCredentials.EXTERNAL_ACCOUNT_AUTHORIZED_USER_FILE_TYPE,
+            ImpersonatedCredentials.IMPERSONATED_CREDENTIALS_FILE_TYPE));
+  }
+
+  /**
+   * Returns an instance of GoogleCredentials defined by JSON
+   *
+   * <p>To be used to parse common credentials fields
+   *
+   * @param json a map from the JSON representing the credentials.
+   * @return the credentials defined by the JSON.
+   * @throws IOException if the credential cannot be created from the JSON.
+   */
+  @VisibleForTesting
+  static GoogleCredentials fromJson(Map<String, Object> json) throws IOException {
+    Builder credentialsBuilder = GoogleCredentials.newBuilder();
+
+    // parse fields that are common for all the types of GoogleCredentials
+    String universeDomain = (String) json.get("universe_domain");
+    if (universeDomain != null
+        && universeDomain.trim().isEmpty()
+        && !universeDomain.equalsIgnoreCase(GoogleAuthUtils.GOOGLE_DEFAULT_UNIVERSE)) {
+      credentialsBuilder.setUniverseDomain(universeDomain);
+    }
+
+    return credentialsBuilder.build();
+  }
+
+  /**
+   * Creates a credential with the provided universe domain.
+   *
+   * @param universeDomain the universe domain to set on the credential. Empty or null value will
+   *     result in the default universe domain 'googleapis.com'
+   * @return credential with the provided universe domain
```

Comment:
```suggestion
   * @return credential with the provided universe domain.
```

Reply:
Periods are too often omitted for single-sentence @params and @returns in this library. I don't thinks its too bad as the don't look like normal sentences when starts with keywords. Let's not enforce it until we clean up significant part of the existing code with a separate PR if we decide to do so. 

For the rest of the docs - Lets have correct sentences.